Library/Homebrew/formula.rb

```diff
@@ -405,11 +413,19 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |pn|
+      pn.basename.to_s.start_with?("HEAD")
```

Comment:
@xu-cheng but we'll need different cleanup behaviour, because we don't know the reason why `latest_head_prefix` returns `nil` and whether it's because no head prefixes exist or it's because all of them are outdated. So we need either to check if there is at least one `HEAD` installed or make `latest_head_prefix` like this:

```ruby
def latest_head_prefix
  head_vesrions = Hash.new { |hash, key| hash[key] = [] }
  installed_prefxies.each do |pn|
    pn_pkgversion = PkgVersion.parse(pn.basename)
    next unless pn_pkgversion.head?
    head_versions[Tab.for_keg(pn).source_modified_time] << pn_pkgversion
  end

  latest_source_modified_time = head_versions.keys.max
  latest_head_version = head_versions[latest_source_modified_time].max_by do |pkg_version|
    pkg_version.revision
  end
  prefix(latest_head_version.first) if latest_head_version
end
```